CounterStrike/src/dev/danablend/counterstrike/CounterStrike.java

```diff
@@ -779,8 +831,18 @@ public CSPlayer getCSPlayer(Player player, boolean create, String colour) {
         }
 
         if (create) {
+            CSPlayer csp = new CSPlayer(this, player, colour);
+
+            //online filters NPCs
+            if (player.isOnline() && (ResourseHash.get(player.getName() + "RES") == null || ResourseHash.get(player.getName() + "RES") == "DEFAULT")) {
+                ResourseHash.remove(player.getName() + "RES");
+                ResourseHash.put(player.getName() + "RES", "QUALITY");

```

Comment:
There is no need to first remove and then use put. Put already will overwrite item with the same key.